pgmq-extension/test/expected/fifo_tests.out

```diff
@@ -0,0 +1,1296 @@
+ send 
+------
+    2
+(1 row)
+
+SELECT * FROM pgmq.send('fifo_test_queue', '{"message": "third"}'::jsonb, '{"x-pgmq-fifo": "group1"}'::jsonb);
+ send 
+------
+    3
+(1 row)
+
+-- Set expected message IDs
+\set msg_id1 1
+\set msg_id2 2
+\set msg_id3 3
+-- Verify we have 3 messages in queue
+SELECT COUNT(*) = 3 FROM pgmq.q_fifo_test_queue;
+ ?column? 
+----------
+ t
+(1 row)
+
+-- Should return only the first message
+SELECT COUNT(*) = 1 FROM pgmq.read_fifo('fifo_test_queue', 10, 5);
+ ?column? 
+----------
+ t
+(1 row)
+
+SELECT msg_id, message, headers FROM pgmq.read_fifo('fifo_test_queue', 10, 5);
+ msg_id | message | headers 
+--------+---------+---------
+(0 rows)
+
+SELECT msg_id = :msg_id1 FROM pgmq.read_fifo('fifo_test_queue', 10, 5);
+ ?column? 
+----------
+(0 rows)
+
+-- Should return no messages (first message still being processed)
+SELECT COUNT(*) = 0 FROM pgmq.read_fifo('fifo_test_queue', 10, 5);
+ ?column? 
+----------
+ t
+(1 row)
+
+SELECT msg_id, message, headers FROM pgmq.read_fifo('fifo_test_queue', 10, 5);
+ msg_id | message | headers 
+--------+---------+---------
+(0 rows)
+
+-- Delete the first message to allow second to be processed
+SELECT * FROM pgmq.delete('fifo_test_queue', :msg_id1);
+ delete 
+--------
+ t
+(1 row)
+
+-- Should now return the second message
+SELECT COUNT(*) = 1 FROM pgmq.read_fifo('fifo_test_queue', 10, 5);
+ ?column? 
+----------
+ t
+(1 row)
+
+SELECT msg_id, message, headers FROM pgmq.read_fifo('fifo_test_queue', 10, 5);
+ msg_id | message | headers 
+--------+---------+---------
+(0 rows)
+
+SELECT msg_id = :msg_id2 FROM pgmq.read_fifo('fifo_test_queue', 10, 5);
+ ?column? 
+----------
+(0 rows)
+
+-- Clean up for next test
+SELECT * FROM pgmq.purge_queue('fifo_test_queue');
+ purge_queue 
+-------------
+           2
+(1 row)
+
+-- test_fifo_multiple_groups
+-- Test 2: Multiple FIFO groups processed in parallel
+SELECT * FROM pgmq.send('fifo_test_queue', '{"message": "group1_msg1"}'::jsonb, '{"x-pgmq-fifo": "group1"}'::jsonb);
+ send 
+------
+    4
+(1 row)
+
+SELECT * FROM pgmq.send('fifo_test_queue', '{"message": "group2_msg1"}'::jsonb, '{"x-pgmq-fifo": "group2"}'::jsonb);
+ send 
+------
+    5
+(1 row)
+
+SELECT * FROM pgmq.send('fifo_test_queue', '{"message": "group1_msg2"}'::jsonb, '{"x-pgmq-fifo": "group1"}'::jsonb);
+ send 
+------
+    6
+(1 row)
+
+SELECT * FROM pgmq.send('fifo_test_queue', '{"message": "group2_msg2"}'::jsonb, '{"x-pgmq-fifo": "group2"}'::jsonb);
+ send 
+------
+    7
+(1 row)
+
+-- Set expected message IDs for this test
+\set msg_id4 4
+\set msg_id5 5
+\set msg_id6 6
+\set msg_id7 7
+-- Verify we have 4 messages in queue
+SELECT COUNT(*) = 4 FROM pgmq.q_fifo_test_queue;
+ ?column? 
+----------
+ t
+(1 row)
+
+-- Should return first message from both groups
+SELECT COUNT(*) = 2 FROM pgmq.read_fifo('fifo_test_queue', 10, 5);
+ ?column? 
+----------
+ t
+(1 row)
```

Comment:
Shouldn't it "wrap" and get more messages from the groups to fill the batch if there's not enough groups to get one of each? 🤔